Code/src/main/java/nl/utwente/group10/ui/components/anchors/ConnectionAnchor.java

```diff
@@ -114,17 +206,24 @@ public boolean hasConnection() {
     public boolean isPrimaryConnected() {
         return isConnected(0);
     }
+    
+    /**
+     * @return True if the primary connection is connected, and its types match.
+     */
+    public boolean isPrimaryConnectedCorrect() {
+        return isPrimaryConnected() && BackendUtils.typesMatch(getSignature().getFresh(), getType().getFresh());
```

Comment:
I do not immediately see what this does, but I do know that this is indeed not the right way around. The front end should never have to use the ```getFresh``` method, and shoudn't. Using this method throws away a lot of useful information about types. Also, as said above, the front end should not have to check whether types do match.